### PR TITLE
Add govulncheck to Go Tests workflow

### PR DESCRIPTION
**Pull request by jrmccluskey**
Adds a [govulncheck](https://pkg.go.dev/golang.org/x/vuln@v1.0.1/cmd/govulncheck) step to the Go Tests workflow to add an extra check against known vulnerabilities. Also allows us to manually trigger the workflow

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by jrmccluskey** [comment]
Waiting on the workflow to run on my local branch to make sure it's happy, do not merge yet

**Comment by jrmccluskey** [comment]
Example Output on my fork:

```
Run go install golang.org/x/vuln/cmd/govulncheck@latest
go: downloading golang.org/x/vuln v1.0.1
go: downloading golang.org/x/mod v0.12.0
go: downloading golang.org/x/tools v0.12.1-0.20230815132531-74c255bcf84[6](https://github.com/jrmccluskey/beam/actions/runs/6027684665/job/16353404191#step:10:7)
Scanning your code and [7](https://github.com/jrmccluskey/beam/actions/runs/6027684665/job/16353404191#step:10:8)[8](https://github.com/jrmccluskey/beam/actions/runs/6027684665/job/16353404191#step:10:9)3 packages across 85 dependent modules for known vulnerabilities...

No vulnerabilities found.

Share feedback at https://go.dev/s/govulncheck-feedback.
```

File: .github/workflows/go_tests.yml [hunk 1]
```diff
@@ -65,3 +66,8 @@ jobs:
           go install "honnef.co/go/tools/cmd/staticcheck@2023.1.3"
           cd sdks/go/pkg/beam
           $(go env GOPATH)/bin/staticcheck ./...
+      - name: Run govulncheck
+        run: |
+          go install golang.org/x/vuln/cmd/govulncheck@latest
+          cd sdks/go/pkg/beam
```
Comment (lostluck) [inline .github/workflows/go_tests.yml:72]: Does this fail if set at sdks/go? We are also concerned with vulnerabilities in the container boot loaders, and this isn't covered.

Ideally we also check 'sdks/java/container/*', 'sdks/python/container/*', 'sdks/typescript/container/*'

Comment (jrmccluskey) [reply .github/workflows/go_tests.yml:72]: It does not fail, we can still cover bootloader tooling. I can change that. We could also invoke at the sdk/ directory and potentially cover the entire code base here

**Comment by damccorm** [comment]
R: @lostluck

**Review by lostluck — APPROVED** [review]

**Pull request merged**